core/src/main/scala/scalaz/Order.scala

```diff
@@ -83,22 +85,32 @@ object Order {
       override def iso: F <=> G = D
     }
 
-  ////
+  implicit val orderDecidableInstance: Decidable[Order] = new OrderDecidableInstance
 
-  implicit val orderInstance: Divisible[Order] = new Divisible[Order] {
-    override def contramap[A, B](r: Order[A])(f: B => A) = r.contramap(f)
+  private[scalaz] class OrderDecidableInstance extends Decidable[Order] {
```

Comment:
this should be mixed in with the `Divisible`.

For performance reasons I think it makes a lot of sense to override `equal` too if it is not already pulled in by inheriting from the `Equal[Divisible]`